omaha/protocol.go

```diff
@@ -0,0 +1,263 @@
+// Copyright 2013-2015 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Google's Omaha application update protocol, version 3.
+//
+// Omaha is a poll based protocol using XML. Requests are made by clients to
+// check for updates or report events of an update process. Responses are given
+// by the server to provide update information, if any, or to simply
+// acknowledge the receipt of event status.
+//
+// https://github.com/google/omaha/blob/wiki/ServerProtocol.md
+package omaha
+
+import (
+	"encoding/xml"
+
+	"github.com/coreos/mantle/version"
+)
+
+// Request sent by the Omaha client
+type Request struct {
+	XMLName       xml.Name      `xml:"request" json:"-"`
+	OS            *OS           `xml:"os"`
+	Apps          []*AppRequest `xml:"app"`
+	Protocol      string        `xml:"protocol,attr"`
+	Version       string        `xml:"version,attr,omitempty"`
+	IsMachine     string        `xml:"ismachine,attr,omitempty"`
+	RequestId     string        `xml:"requestid,attr,omitempty"`
+	SessionId     string        `xml:"sessionid,attr,omitempty"`
+	UserId        string        `xml:"userid,attr,omitempty"`
+	InstallSource string        `xml:"installsource,attr,omitempty"`
+	TestSource    string        `xml:"testsource,attr,omitempty"`
+
+	// update engine extension, duplicates the version attribute.
+	UpdaterVersion string `xml:"updaterversion,attr,omitempty"`
+}
+
+func NewRequest() *Request {
+	return &Request{
+		Protocol: "3.0",
+		Version:  version.Version,
```

Comment:
Should this be dis-entangled from mantle versioning? I see we are not vedoring it here, so I think this and the import at the top just slipped in.